lib/Geo/GeoJson.h

```diff
@@ -52,58 +63,121 @@ enum class Type : uint8_t {
   POLYGON,
   MULTI_POINT,
   MULTI_LINESTRING,
-  MULTI_POLYGON,  // todo
-  GEOMETRY_COLLECTION
+  MULTI_POLYGON,
+  // TODO(MBkkt) we don't support GeometryCollection so it's a having it in Type
+  //  a very little slowdown parsing, but provide better error.
+  //
+  //  Also I think we can support it, because for all operations with geometry
+  //  we use S2ShapeIndex and it support construction from collection of
+  //  S2Region so I think it's nice feature.
+  GEOMETRY_COLLECTION,
 };
 
-struct Fields {
-  static constexpr auto kCoordinates =
-      "coordinates";                     // mandatory, value depends on type
-  static constexpr auto kType = "type";  // mandatory
-};
+/// About without validation, it should be used when we parse data from our
+/// indexes. Because this checks very expensive and we don't want to make it in
+/// every query. Instead of this we store in our indexes only valid data.
+/// @note
+/// In other words I make assumption if Validation is false all parsing
+/// is successful.
+/// @note
+/// In Maintainer mode under x86 linux Validation is false,
+/// use same code as Validation is true. But make TRI_ASSERT that result is ok.
+/// @note
+/// For all functions without Validation parameter it's implicitly equal true.
+/// @note
+/// For all functions without legacy parameter it's implicitly equal false.
```

Comment:
> Why can we afford this? How can we ensure that only data, which has previously been parsed successfully, is parsed without validation?

Because it was in code. We use this function only in some tests and Functions.cpp

Reply:
> Why can we afford this? How can we ensure that only data, which has previously been parsed successfully, is parsed without validation?

Or I don't understand your questions, can you explain what do you mean?

In general this comments just clarification, not behavior changes.

We have one parameters for all functions:
Validation, when Validation true, data will be checked, if Validation false we assume it's correct (because when we insert it we already checked it, and insert only Valid data, at least for arangosearch)

Legacy, it's only about polygons (parsePolygon, parseMultiPolygon, parseRegion), first two functions used only with legacy=false, so I changed they, now they don't have such parameter